File: Julia_exercises/gc_content.jl
trna="AAGGGCTTAGCTTAATTAAAGTGGCTGATTTGCGTTCAGTTGATGCAGAGTGGGGTTTTGCAGTCCTTA"
#Check length of sequence
trna_length = length(trna) 
g_count = count("G",trna)
c_count = count("C",trna)
t_count = count("T",trna)
a_count = count("A",trna)
#Calculating gc_count
percentage_gc= (g_count+c_count)/trna_length
percentage_at = (a_count+ t_count)/trna_length

#Display the percentage at & gc content 
println("The percentage gc content is , $percentage_gc")
println("The percentage at content is ", percentage_at)

File: Julia_exercises/gc_content.py
#Calculating gc content of a sequence trna from ncbi
#Let's use the mitochondrial tRNA (NCBI Reference Sequence: NC_012920.1) to practice with string operations: NCBI Reference Sequence: NC_012920.1

trna="AAGGGCTTAGCTTAATTAAAGTGGCTGATTTGCGTTCAGTTGATGCAGAGTGGGGTTTTGCAGTCCTTA"

trna_length = len(trna)
g_count = trna.count('G')
c_count = trna.count('C')
a_count = trna.count('A')
t_count = trna.count('T')
percentage_gc = (g_count+c_count)/trna_length * 100
percentage_at = (a_count+ t_count)/trna_length *100
#Dispaly the percentage gc & at content
print("The percentage gc content of trna is",percentage_gc)
print("The percentage at conntent of trna sequence is",percentage_at)

#String manipulation
amino_acid = "MNKMDLVADVAEKTDLSKAKATEVIDAVFA"
amino_acid[0]
amino_acid[-1]
amino_acid[4]

#The above amino acid is a bacterial restriction enzyme that recognizes "TCCGGA". 
# Find the first restriction site in the following sequence: AAAAATCCCGAGGCGGCTATATAGGGCTCCGGAGGCGTAATATAAAA
sequence = "AAAAATCCCGAGGCGGCTATATAGGGCTCCGGAGGCGTAATATAAAA"
sequence.find("TCCGGA")

#Data structures

